components/org.apache.stratos.rest.endpoint/src/main/java/org/apache/stratos/rest/endpoint/api/StratosApiV41Utils.java

```diff
@@ -2836,52 +2838,58 @@ public static void updateExistingTenant(org.apache.stratos.common.beans.TenantIn
         } catch (UserStoreException e) {
             String msg = "Error in retrieving the tenant id for the tenant domain: " + tenantDomain + ".";
             log.error(msg, e);
-            throw new Exception(msg, e);
+            throw new RestAPIException(msg, e);
         }
 
         Tenant tenant;
         try {
             tenant = (Tenant) tenantManager.getTenant(tenantId);
         } catch (UserStoreException e) {
-            String msg = "Error in retrieving the tenant id for the tenant domain: " +
-                    tenantDomain + ".";
+            String msg = "Error in retrieving the tenant from tenant id: " +
+                    tenantId + ".";
             log.error(msg, e);
-            throw new TenantNotFoundException(msg, e);
+            throw new RestAPIException(msg, e);
         }
 
         // filling the first and last name values
-        if (tenantInfoBean.getFirstname() != null && !tenantInfoBean.getFirstname().trim().equals("")) {
+        if (StringUtils.isBlank(tenantInfoBean.getFirstname())) {
             try {
                 CommonUtil.validateName(tenantInfoBean.getFirstname(), "First Name");
             } catch (Exception e) {
                 String msg = "Invalid first name is provided.";
                 log.error(msg, e);
-                throw new Exception(msg, e);
+                throw new RestAPIException(msg, e);
             }
         }
-        if (tenantInfoBean.getLastname() != null && !tenantInfoBean.getLastname().trim().equals("")) {
+        if (StringUtils.isBlank(tenantInfoBean.getLastname())) {
             try {
                 CommonUtil.validateName(tenantInfoBean.getLastname(), "Last Name");
             } catch (Exception e) {
```

Comment:
Better to catch with the specific exceptions rather than catching whole exception